cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/config/ConfigurationHolder.java

```diff
@@ -45,13 +57,46 @@ public TemplatesConfiguration readTemplatesConfiguration(Trigger trigger) {
         if (!templatesConfigurations.containsKey(templateFolder)) {
             templatesConfigurations.put(templateFolder, Maps.<String, TemplatesConfiguration> newHashMap());
 
-            TemplatesConfiguration config = new TemplatesConfiguration(configurationPath, trigger);
+            TemplatesConfiguration config = new TemplatesConfiguration(configurationPath, trigger, this);
             templatesConfigurations.get(templateFolder).put(trigger.getId(), config);
         }
 
         return templatesConfigurations.get(templateFolder).get(trigger.getId());
     }
 
+    /**
+     * Reads the {@link TemplatesConfiguration} from cache or from file if not present in cache, just storing
+     * the external increment we want.
+     * @param trigger
+     *            to get matcher declarations from
+     * @param incrementToSearch
+     *            name of the increment to search
+     * @return the {@link TemplatesConfiguration}
+     * @throws InvalidConfigurationException
+     *             if the configuration is not valid
+     */
+    public TemplatesConfiguration readTemplatesConfiguration(Trigger trigger, String incrementToSearch) {
```

Comment:
It seems, that it is just possible to reference increments. Can you check that?
What about templateScans and templates to be referenced?
Maybe the code architecture has to rethought to not add multiple parameters per case.

Reply:
According to the documentaiton @jdiazgon created, this is the only thing that's possible so far, i'll have a look into the others as well